Pipeline/tests/PipelineFileInputTests.py

```diff
@@ -0,0 +1,238 @@
+            expected_response = json.load(f)
+
+        self.assertEqual(200, response.status_code)
+        self.compare_core(expected_response, response.json)
+        self.assertTrue("rnamigos_result" not in response.json)
+        self.assertTrue("similar_motifs" in response.json)
+
+    @patch('requests.request')
+    def test_failure_bp(self, mock_post):
+        '''
+        Tests for a failure case where a failure occurred in BayesPairing
+        '''
+        with open(os.path.join(CURRENT_DIRECTORY, "responses/mock_responses/RESPONSE_BP_ERROR.json")) as f:
+            bp_response = json.load(f)
+
+        mock_post.return_value = MockResponse(bp_response, 400)
+
+        payload = FILE_INPUT_FULL
+        headers = {}
+
+        with open(os.path.join(CURRENT_DIRECTORY, "data/INVALID_FASTA.fa"), mode="rb") as f:
+            payload["bp_input"] = (io.BytesIO(f.read()), "INPUT_SEQUENCE.fa")
+
+        response = self.app.post(
+            PL_FILE_URL, content_type='multipart/form-data', headers=headers, data=payload)
+
+        self.assertEqual(400, response.status_code)
+
+    @patch('requests.request')
+    def test_failure_vernal(self, mock_post):
+        '''
+        Tests for a failure case where a failure occurred in VeRNAl
+        '''
+        with open(os.path.join(CURRENT_DIRECTORY, "responses/mock_responses/RESPONSE_VERNAL_ERROR.json")) as f:
+            vernal_response = json.load(f)
+
+        mock_post.return_value = MockResponse(vernal_response, 400)
+
+        payload = FILE_INPUT_FULL
+        headers = {}
+
+        with open(os.path.join(CURRENT_DIRECTORY, "data/INPUT_SEQUENCE.fa"), mode="rb") as f:
+            payload["bp_input"] = (io.BytesIO(f.read()), "INPUT_SEQUENCE.fa")
+
+        response = self.app.post(
+            PL_FILE_URL, content_type='multipart/form-data', headers=headers, data=payload)
+
+        self.assertEqual(400, response.status_code)
+
+    @patch('requests.request')
+    def test_failure_rnamigos(self, mock_post):
+        '''
+        Tests for a failure case where a failure occurred in RNAmigos
+        '''
+        with open(os.path.join(CURRENT_DIRECTORY, "responses/mock_responses/RESPONSE_RNAMIGOS_ERROR.json")) as f:
+            rnamigos_response = json.load(f)
+
+        mock_post.return_value = MockResponse(rnamigos_response, 400)
+
+        payload = FILE_INPUT_FULL
+        headers = {}
+
+        with open(os.path.join(CURRENT_DIRECTORY, "data/INPUT_SEQUENCE.fa"), mode="rb") as f:
+            payload["bp_input"] = (io.BytesIO(f.read()), "INPUT_SEQUENCE.fa")
+
+        response = self.app.post(
+            PL_FILE_URL, content_type='multipart/form-data', headers=headers, data=payload)
+
+        self.assertEqual(400, response.status_code)
+
+    @patch('requests.request')
+    def test_failure_no_arguments(self, mock_post):
+        '''
+        Tests for a failure case where an no arguments are provided
```

Comment:
same comment as above, and maybe a test where 2/3 tools fail with different errors. How would that case be handled and which code would be returned to the client?

Reply:
if 1 tool fails, an error is returned